pyrobosim/pyrobosim/core/world.py

```diff
@@ -1742,14 +1744,23 @@ def is_connectable(
 
         # Check the occupancy of all the test points.
         for x_check, y_check in zip(x_pts[1:], y_pts[1:]):
-            if self.check_occupancy(Pose(x=x_check, y=y_check)):
+            if self.check_occupancy(
+                Pose(x=x_check, y=y_check),
+                partial_observability_hallway_states,
+                recorded_closed_hallways,
+            ):
                 return False
 
         # If the loop was traversed for all points without returning, we can
         # connect the points.
         return True
 
-    def check_occupancy(self, pose: Pose | Sequence[float]) -> bool:
+    def check_occupancy(
```

Comment:
I'm starting to think if at this point it might not be easier to create a separate utility function that accepts the world and the robot, and grabs all the extra information directly from these objects?